In [1]:
# 1. Name the car which is having the selling price more than 600000.
# 2. Name the car having manual transmission and also find how much km its driven.
# 3. Name the list of the car having sold by first owner.
# 4. Name the list of car having selling price more than 10lac but manufacturing
# year will be 2010.
# 5. Find the most expensive car.
# 6. Find the most cheapest car.
# 7.Find tthe car which is max driven.
# 8. Find the car which is min driven.
# 9. What is the fuel type & transmission of "Maruti Car".
# 10.What is the selling price of 'Toyota Corolla Altis 1.8 VL CVT' car.

 


In [2]:
import pandas as pd

In [3]:
df=pd.read_csv('CAR DETAILS FROM CAR DEKHO.csv')

In [4]:
df.columns

Index(['name', 'year', 'selling_price', 'km_driven', 'fuel', 'seller_type',
       'transmission', 'owner'],
      dtype='object')

In [5]:
df.isnull().sum()

name             0
year             0
selling_price    0
km_driven        0
fuel             0
seller_type      0
transmission     0
owner            0
dtype: int64

In [6]:
df.head()

name  year  selling_price  km_driven    fuel  \
0             Maruti 800 AC  2007          60000      70000  Petrol   
1  Maruti Wagon R LXI Minor  2007         135000      50000  Petrol   
2      Hyundai Verna 1.6 SX  2012         600000     100000  Diesel   
3    Datsun RediGO T Option  2017         250000      46000  Petrol   
4     Honda Amaze VX i-DTEC  2014         450000     141000  Diesel   

  seller_type transmission         owner  
0  Individual       Manual   First Owner  
1  Individual       Manual   First Owner  
2  Individual       Manual   First Owner  
3  Individual       Manual   First Owner  
4  Individual       Manual  Second Owner

In [7]:
df.tail()

name  year  selling_price  km_driven  \
4335  Hyundai i20 Magna 1.4 CRDi (Diesel)  2014         409999      80000   
4336           Hyundai i20 Magna 1.4 CRDi  2014         409999      80000   
4337                  Maruti 800 AC BSIII  2009         110000      83000   
4338     Hyundai Creta 1.6 CRDi SX Option  2016         865000      90000   
4339                     Renault KWID RXT  2016         225000      40000   

        fuel seller_type transmission         owner  
4335  Diesel  Individual       Manual  Second Owner  
4336  Diesel  Individual       Manual  Second Owner  
4337  Petrol  Individual       Manual  Second Owner  
4338  Diesel  Individual       Manual   First Owner  
4339  Petrol  Individual       Manual   First Owner

In [8]:
df.sample(5)

name  year  selling_price  km_driven    fuel  \
4050    Maruti Wagon R LXI Minor  2008         120000     110000  Petrol   
3073   Hyundai i20 Asta 1.4 CRDi  2013         300000      70000  Diesel   
1744      Mahindra XUV500 W6 2WD  2016         900000      47000  Diesel   
3597      Hyundai i20 1.4 Sportz  2017         740000      60000  Diesel   
1905  Ford Figo 1.5D Titanium MT  2017         600000      35122  Diesel   

     seller_type transmission         owner  
4050  Individual       Manual   Third Owner  
3073  Individual       Manual  Second Owner  
1744  Individual       Manual   First Owner  
3597  Individual       Manual   First Owner  
1905      Dealer       Manual   First Owner

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4340 entries, 0 to 4339
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   name           4340 non-null   object
 1   year           4340 non-null   int64 
 2   selling_price  4340 non-null   int64 
 3   km_driven      4340 non-null   int64 
 4   fuel           4340 non-null   object
 5   seller_type    4340 non-null   object
 6   transmission   4340 non-null   object
 7   owner          4340 non-null   object
dtypes: int64(3), object(5)
memory usage: 271.4+ KB


In [10]:
df.describe()

year  selling_price      km_driven
count  4340.000000   4.340000e+03    4340.000000
mean   2013.090783   5.041273e+05   66215.777419
std       4.215344   5.785487e+05   46644.102194
min    1992.000000   2.000000e+04       1.000000
25%    2011.000000   2.087498e+05   35000.000000
50%    2014.000000   3.500000e+05   60000.000000
75%    2016.000000   6.000000e+05   90000.000000
max    2020.000000   8.900000e+06  806599.000000

# observation:  no data is missing in the dataset or having the null value

In [11]:
df.corr()

year  selling_price  km_driven
year           1.000000       0.413922  -0.419688
selling_price  0.413922       1.000000  -0.192289
km_driven     -0.419688      -0.192289   1.000000

In [12]:
# 1. Name the car which is having the selling price more than 600000.

df[df['selling_price']>600000]['name']

8                 Hyundai Creta 1.6 VTVT S
12         Toyota Corolla Altis 1.8 VL CVT
21                Hyundai Creta 1.6 VTVT S
25         Toyota Corolla Altis 1.8 VL CVT
27             Hyundai Venue SX Opt Diesel
                       ...                
4311               Toyota Camry Hybrid 2.5
4312                 Maruti Ertiga 1.5 VDI
4313    Ford Endeavour 2.2 Titanium AT 4X2
4332          Mahindra Scorpio S2 7 Seater
4338      Hyundai Creta 1.6 CRDi SX Option
Name: name, Length: 1057, dtype: object

# obervation: this dataset contains 1057 cars which are having the selling price more than 600000

In [13]:
# 2. Name the car having manual transmission and also find how much km its driven.

df[df['transmission']=='Manual'][['name','km_driven']]

name  km_driven
0                           Maruti 800 AC      70000
1                Maruti Wagon R LXI Minor      50000
2                    Hyundai Verna 1.6 SX     100000
3                  Datsun RediGO T Option      46000
4                   Honda Amaze VX i-DTEC     141000
...                                   ...        ...
4335  Hyundai i20 Magna 1.4 CRDi (Diesel)      80000
4336           Hyundai i20 Magna 1.4 CRDi      80000
4337                  Maruti 800 AC BSIII      83000
4338     Hyundai Creta 1.6 CRDi SX Option      90000
4339                     Renault KWID RXT      40000

[3892 rows x 2 columns]

# observation: dataset contains 3892 data which are having there transmission type as 'Manual'

In [14]:
# 3. Name the list of the car having sold by first owner.

df[df['owner']=='First Owner']['name']

0                                      Maruti 800 AC
1                           Maruti Wagon R LXI Minor
2                               Hyundai Verna 1.6 SX
3                             Datsun RediGO T Option
5                               Maruti Alto LX BSIII
                            ...                     
4324                            Maruti Alto 800 Base
4332                    Mahindra Scorpio S2 7 Seater
4334    Toyota Innova 2.5 VX (Diesel) 8 Seater BS IV
4338                Hyundai Creta 1.6 CRDi SX Option
4339                                Renault KWID RXT
Name: name, Length: 2832, dtype: object

# observation: only 2832 owner was there who sold there car as first hand used

In [15]:
# Name the list of car having selling price more than 10lac but manufacturing
# # year will be 2010.

df[(df['selling_price']>1000000) & (df['year']==2010)]

name  year  selling_price  \
163                  Jaguar XJ 5.0 L V8 Supercharged  2010        2550000   
2685                        Audi Q5 2.0 TFSI Quattro  2010        1100000   
3782                      Toyota Fortuner 3.0 Diesel  2010        1250000   
3875  Land Rover Range Rover 4.4 Diesel LWB Vogue SE  2010        4200000   

      km_driven    fuel seller_type transmission         owner  
163       40000  Petrol  Individual    Automatic  Second Owner  
2685     110000  Petrol  Individual    Automatic   First Owner  
3782     205000  Diesel  Individual       Manual  Second Owner  
3875     100000  Diesel      Dealer    Automatic   First Owner

In [16]:
# 5. Find the most expensive car.

df[df['selling_price']==df['selling_price'].max()]['name']

3872    Audi RS7 2015-2019 Sportback Performance
Name: name, dtype: object

In [17]:
# 6. Find the most cheapest car.

df[df['selling_price']==df['selling_price'].min()]['name']

2662    Ford Ikon 1.6 ZXI NXt
Name: name, dtype: object

In [18]:
# 7.Find tthe car which is max driven.

df[df['km_driven']==df['km_driven'].max()]['name']

1243    Maruti Swift VXI BSIII
Name: name, dtype: object

In [19]:
# 8. Find the car which is min driven.

df[df['km_driven']==df['km_driven'].min()]['name']

1312    Mahindra Quanto C6
Name: name, dtype: object

In [20]:
# 9. What is the fuel type & transmission of "Maruti Car".

df[df['name']==df['name'][0]][['fuel','transmission']]

fuel transmission
0     Petrol       Manual
13    Petrol       Manual
175   Petrol       Manual
259   Petrol       Manual
372   Petrol       Manual
402   Petrol       Manual
669   Petrol       Manual
1284  Petrol       Manual
1446  Petrol       Manual
1726  Petrol       Manual
3181  Petrol       Manual
3215  Petrol       Manual
3217  Petrol       Manual
3259  Petrol       Manual
3285  Petrol       Manual
3314  Petrol       Manual
3503  Petrol       Manual
3630  Petrol       Manual
3652  Petrol       Manual
3740  Petrol       Manual
3856  Petrol       Manual
4088  Petrol       Manual
4323  Petrol       Manual

In [21]:
# 10.What is the selling price of 'Toyota Corolla Altis 1.8 VL CVT' car.

df[df['name']=='Toyota Corolla Altis 1.8 VL CVT']['selling_price']

12    1650000
25    1650000
Name: selling_price, dtype: int64

In [22]:
# how many car is automatic

df[df['transmission']=='Automatic'].shape[0]

448

In [23]:
# how many car is manual

df[df['transmission']=='Manual'].shape[0]

3892

In [24]:
#how many car are solved by dealer

df[df['seller_type']=='Dealer'].shape[0]

994

In [25]:
#how many car sold by individual

df[df['seller_type']=='Individual'].shape[0]

3244

In [26]:
#how many car whose petrol is petrol

df[df['fuel']=='Petrol'].shape[0]

2123

In [27]:
#how many car whose fuel is diesel

df[df['fuel']=='Diesel'].shape[0]

2153

In [28]:
# hwo many car whose fuel is CNG

df[df['fuel']=='CNG'].shape[0]

40

In [29]:
#how many car of petrol is max km_driven

df[df['fuel']=='Petrol'][['name','km_driven','fuel']].max()

name         Volkswagen Vento Petrol Highline AT
km_driven                                 806599
fuel                                      Petrol
dtype: object

In [30]:
#how many car of petrol is min km_driven

df[df['fuel']=='Petrol'][['name','km_driven','fuel']].min()

name         Ambassador Grand 1800 ISZ MPFI PW CL
km_driven                                     101
fuel                                       Petrol
dtype: object

In [31]:
#how many car of deisel is max km_driven

df[df['fuel']=='Diesel'][['name','km_driven','fuel']].max()

name         Volvo XC60 D5 Inscription
km_driven                       560000
fuel                            Diesel
dtype: object

In [32]:
#how many car of deisel is min km_driven

df[df['fuel']=='Diesel'][['name','km_driven','fuel']].min()

name         Ambassador CLASSIC 1500 DSL AC
km_driven                                 1
fuel                                 Diesel
dtype: object

In [33]:
#how many car of CNG is max km_driven

df[df['fuel']=='CNG'][['name','km_driven','fuel']].max()

name         Tata Indigo CS Emax CNG GLX
km_driven                         120000
fuel                                 CNG
dtype: object

In [34]:
#how many car of CNG is min km_driven

df[df['fuel']=='CNG'][['name','km_driven','fuel']].min()

name         Chevrolet Aveo 1.4 CNG
km_driven                      4000
fuel                            CNG
dtype: object

In [35]:
df.columns

Index(['name', 'year', 'selling_price', 'km_driven', 'fuel', 'seller_type',
       'transmission', 'owner'],
      dtype='object')

In [37]:
df.count()

name             4340
year             4340
selling_price    4340
km_driven        4340
fuel             4340
seller_type      4340
transmission     4340
owner            4340
dtype: int64

In [42]:
df['name'].unique()

array(['Maruti 800 AC', 'Maruti Wagon R LXI Minor',
       'Hyundai Verna 1.6 SX', ..., 'Mahindra Verito 1.5 D6 BSIII',
       'Toyota Innova 2.5 VX (Diesel) 8 Seater BS IV',
       'Hyundai i20 Magna 1.4 CRDi'], dtype=object)

In [43]:
df['name'].nunique()

1491

In [45]:
df.value_count()

AttributeError: 'DataFrame' object has no attribute 'value_count'